integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r4/PatientDomainTest.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.providers.r4;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.equalToIgnoringCase;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.Assert.assertThrows;
+
+import java.util.ArrayList;
+import java.util.UUID;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.rest.api.MethodOutcome;
+import ca.uhn.fhir.rest.client.api.IGenericClient;
+import ca.uhn.fhir.rest.client.interceptor.BasicAuthInterceptor;
+import ca.uhn.fhir.rest.server.exceptions.ResourceGoneException;
+import ca.uhn.fhir.rest.server.exceptions.ResourceNotFoundException;
+import com.google.common.collect.Sets;
+import org.hl7.fhir.r4.model.CodeableConcept;
+import org.hl7.fhir.r4.model.Enumerations;
+import org.hl7.fhir.r4.model.Extension;
+import org.hl7.fhir.r4.model.HumanName;
+import org.hl7.fhir.r4.model.IdType;
+import org.hl7.fhir.r4.model.Identifier;
+import org.hl7.fhir.r4.model.Patient;
+import org.hl7.fhir.r4.model.Reference;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.openmrs.module.fhir2.FhirConstants;
+
+public class PatientDomainTest {
+
+    private static final String SERVER_BASE = "http://localhost:8080/openmrs/ws/fhir2/R4";
+
+    private static final FhirContext FHIR_CONTEXT = FhirContext.forR4();
+
+    private static final IGenericClient CLIENT = FHIR_CONTEXT.newRestfulGenericClient(SERVER_BASE);
+
+    private final String NON_EXISTING_PATIENT_UUID = UUID.randomUUID().toString();
+
+    private static String patientUuid = "34d45fr6-f3a9-4f46-8341-15f0de3d8476";
+
+    private static final String PATIENT_FAMILY_NAME_JAMESON = "Jameson";
+
+    private static final String PATIENT_GIVEN_NAME_JONAH = "Jonah";
+
+    private static final String PATIENT_LOCATION_UUID = "58c57d25-8d39-41ab-8422-108a0c277d98";
+
+    private static final String PATIENT_IDENTIFIER_ID = "1000H7Y";
+
+    private static final String PATIENT_CONCEPT_TEXT = "OpenMRS ID";
+
+    private static final String CREDENTIAL_STRING = "admin:Admin123";
+
+    @BeforeClass
+    public static void authenticate() {
+        FHIR_CONTEXT.getRestfulClientFactory().setConnectTimeout(200 * 1000);
+        CLIENT.registerInterceptor(new BasicAuthInterceptor(CREDENTIAL_STRING));
+    }
+
+    @BeforeClass
```

Comment:
Whats the difference here, having two @beforeClass methods ? , Cant we combine them at one .

Reply:
Ohh @sherrif10 
I am gonna polish code and i will migrate the logic into on `BeforeClass` method